build/AssemblyRedirectionSourceGenerator.cs

```diff
@@ -35,39 +35,48 @@ public static void Generate(string assembliesFolderPath, string generatedFilePat
         Log.Information("Assembly redirection source generated {0}", generatedFilePath);
     }
 
-    static string GenerateSourceContents(SortedDictionary<string, AssemblyNameDefinition> assemblies)
+    private static string GenerateSourceContents(SortedDictionary<string, AssemblyNameDefinition> assemblies)
     {
-        var sb = new StringBuilder(assemblies.Count * 256);
-        sb.AppendLine("""
-    /*
-     * Copyright The OpenTelemetry Authors
-     * SPDX-License-Identifier: Apache-2.0
-     */
-    """);
-        sb.AppendLine();
-        sb.AppendLine($"// Auto-generated file, do not change it - generated by the {nameof(AssemblyRedirectionSourceGenerator)} type");
-        sb.Append(@"
-#include ""cor_profiler.h""
+        return
+        $$"""
+        /*
+         * Copyright The OpenTelemetry Authors
+         * SPDX-License-Identifier: Apache-2.0
+         */
+
+        // Auto-generated file, do not change it - generated by the {{nameof(AssemblyRedirectionSourceGenerator)}} type
+
+        #include "cor_profiler.h"
+
+        #ifdef _WIN32
+        namespace trace
+        {
+        void CorProfiler::InitNetFxAssemblyRedirectsMap()
+        {
+            assembly_version_redirect_map_.insert({
+                {{GenerateEntries(assemblies)}}
+            });
+        }
+        }
+        #endif
+
+        """;
+    }
+
+    private static string GenerateEntries(SortedDictionary<string, AssemblyNameDefinition> assemblies)
+    {
+        var sb = new StringBuilder(assemblies.Count * 77 /* Longest line */);
```

Comment:
not that it matters much but this may make people think that they need to go and look for the longest line if they update this file. I would just put named constant, something like `longLineLength` and give a value like 80.